valhalla/proposals/accounting.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import requests
+from django.conf import settings
+
+from valhalla.proposals.models import Semester
+
+logger = logging.getLogger(__name__)
+
+
+def split_time(start, end, chunks=4):
+    spans = []
+    chunk = (end - start) / chunks
+    spans.append((start, start + chunk))
+    for i in range(0, chunks - 1):
+        spans.append((spans[i][1], spans[i][1] + chunk))
+    return spans
+
+
+def get_time_totals_from_pond(timeallocation, start, end, too, recur=0):
+    """
+    Pond queries are too slow with large proposals and time out, this hack splits the query
+    when a timeout occurs
+    """
+    if recur > 3:
+        raise RecursionError('Pond is timing out, too much recursion.')
+
+    total = 0
+    try:
+        total += query_pond(timeallocation.proposal.id, start, end, timeallocation.telescope_class, too)
+    except requests.HTTPError:
+        logger.warning('We got a pond inception. Splitting further.')
```

Comment:
But this won't stop execution will it ? It will split into more parts which will either also fail to complete or potentially give the wrong (smaller) answer for time used if the POND comes back in time for time sections no. 2 and no. 3 but was down for the likely more heavily used time section no. 1